### PR TITLE
Support OTHER type, Replace IBAN/BIC model fields, ++

### PR DESCRIPTION
**Pull request by beshrkayali**
- Switch from django_iban to localflavor IBAN / BIC fields.
- Minor fix on Bank Account entity creation for IBAN type bank account (Send in a Details object)
- Support `OTHER` bank account type to allow for Bank Accounts in countries with NO IBAN
- Always save country of bank account along with the Details of bank account

### CODE REVIEW

**Comment by appel268576** [comment]
:+1:

File: mangopay/tests/bank_account.py [hunk 1]
```diff
@@ -4,20 +4,51 @@
...
+        id_ = 33322
+        mock_client.return_value = MockMangoPayApi(bank_account_id=id_)
+        self.assertIsNone(self.bank_account_iban.mangopay_id)
+        self.bank_account_iban.create()
+        MangoPayBankAccount.objects.get(id=self.bank_account_iban.id,
+                                        mangopay_id=id_)
+        self.assertEqual(self.bank_account_iban.account_type,
+                         BA_BIC_IBAN)
+        self.assertTrue(self.bank_account_iban.iban is not "")
+        self.assertTrue(self.bank_account_iban.account_number is None)
```
Comment (flaviamissi) [inline mangopay/tests/bank_account.py]: Not very important but maybe you can use more specific assertions here, e.g.  `assertNotEqual("", self.bank_account_iban.iban)`
and
`assertIsNone(self.bank_account_iban.account_number)`

File: mangopay/tests/bank_account.py [hunk 2]
```diff
@@ -4,20 +4,51 @@
...
+        id_ = 22333
+        mock_client.return_value = MockMangoPayApi(bank_account_id=id_)
+        self.assertIsNone(self.bank_account_other.mangopay_id)
+        self.bank_account_other.create()
+        MangoPayBankAccount.objects.get(id=self.bank_account_other.id,
+                                        mangopay_id=id_)
+        self.assertEqual(self.bank_account_other.account_type,
+                         BA_OTHER)
+        self.assertTrue(self.bank_account_other.iban is None)
+        self.assertTrue(self.bank_account_other.account_number is not None)
```
Comment (flaviamissi) [inline mangopay/tests/bank_account.py]: Same here about the assertions ;)

**Comment by flaviamissi** [comment]
I made some comments, after reading them you can merge this!

**Pull request merged**